src/game/Utils/STIConvert.cc

```diff
@@ -127,28 +130,43 @@ void WriteSTIFile(UINT8* const pData, SGPPaletteEntry* const pPalette, const INT
 				STCIPaletteEntry.ubRed   = pSGPPaletteEntry[uiLoop].r;
 				STCIPaletteEntry.ubGreen = pSGPPaletteEntry[uiLoop].g;
 				STCIPaletteEntry.ubBlue  = pSGPPaletteEntry[uiLoop].b;
-				fwrite(&STCIPaletteEntry, sizeof(STCIPaletteEntry), 1, pOutput);
+				if (!File_writeAll(file.get(), reinterpret_cast<const uint8_t*>(&STCIPaletteEntry), sizeof(STCIPaletteEntry)))
+				{
+					RustPointer<char> err(getRustError());
+					SLOGW("WriteSTIFile: %s", err.get());
+				}
 			}
 		}
 		if (Header.fFlags & STCI_ETRLE_COMPRESSED)
 		{
-			fwrite( subImages.data(), subImages.size() * sizeof(STCISubImage), 1, pOutput );
+			if (!File_writeAll(file.get(), reinterpret_cast<const uint8_t*>(subImages.data()), subImages.size() * sizeof(STCISubImage)))
+			{
+				RustPointer<char> err(getRustError());
+				SLOGW("WriteSTIFile: %s", err.get());
+			}
 		}
 	}
 
 	// write file data
 	Assert(Header.fFlags & STCI_ETRLE_COMPRESSED);
 	Assert(pOutputBuffer);
-	fwrite( pOutputBuffer, Header.uiStoredSize, 1, pOutput );
+	if (!File_writeAll(file.get(), reinterpret_cast<const uint8_t*>(pOutputBuffer), Header.uiStoredSize))
+	{
+		RustPointer<char> err(getRustError());
+		SLOGW("WriteSTIFile: %s", err.get());
+	}
 
 	// write app-specific data (blanked to 0)
+	const uint8_t zero = 0;
 	for (uiLoop = 0; uiLoop < Header.uiAppDataSize; uiLoop++)
 	{
-		fputc( 0, pOutput );
+		if (!File_writeAll(file.get(), &zero, sizeof(zero)))
+		{
+			RustPointer<char> err(getRustError());
+			SLOGW("WriteSTIFile: %s", err.get());
```

Comment:
Opening a random commit ... it seems like we could use another helper function or three for all this logging.

Reply:
Maybe after all the `wchar_t` code is converted (#937).

While coding, the `printf` format strings were an annoyance so I will probably target them.
Logging functions are one of those targets.